src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -267,8 +269,20 @@ protected internal int GetInvocationCounter()
         /// <returns>The next random number.</returns>
         protected internal BigInteger GetRandom()
         {
-            nonceData = Cryptography.Helper.Murmur128(nonceData, ProtocolSettings.Network);
-            return new BigInteger(nonceData, isUnsigned: true);
+            byte[] buffer;
+            long price;
+            if (IsHardforkEnabled(Hardfork.HF_Aspidochelone))
+            {
+                buffer = Cryptography.Helper.Murmur128(nonceData, ProtocolSettings.Network + random_times++);
```

Comment:
@erikzhang  We miss to update the `nonceData` again? also, the first iteration it's the same as before, maybe we should test if it produce a fork with the last implementation again.

Reply:
We don't need to update `nonceData` because we have used `random_times`.

> maybe we should test if it produce a fork with the last implementation again.

And this is included in the hard fork. So, yes, it do produce a fork.